### PR TITLE
Do not flow secretness on input values

### PR DESCRIPTION
**Pull request by iwahbe**
Fixes https://github.com/pulumi/pulumi-go-provider/issues/106 (see issue for mechanics of the fix)

### CODE REVIEW

File: infer/tests/create_test.go [hunk 1]
```diff
@@ -88,6 +88,41 @@ func TestCreate(t *testing.T) {
 		}, resp.Properties)
 	})
 
+	t.Run("unwired-secrets", func(t *testing.T) {
```
Comment (mjeffryes) [inline infer/tests/create_test.go:91]: Should we test the condition where the input has the same name, but a different value in the output still creates a secret?

**Review by mjeffryes — COMMENTED** [review]
This looks reasonable to me, but I don't know if I have enough context to fully understand the impact of this change.

File: infer/tests/create_test.go (hunk 1 above)
Comment (iwahbe) [reply infer/tests/create_test.go:91]: We probably should. I'll add that test.

Comment (iwahbe) [reply infer/tests/create_test.go:91]: Added in https://github.com/pulumi/pulumi-go-provider/pull/107/commits/4907e778f086ddaabc7d310e88969e52dd59b02d.

**Review by mjeffryes — APPROVED** [review]

**Pull request merged**